--- filefield_formatter.inc 2010-06-17 02:41:22 +0000
+++ filefield_formatter.inc 2010-06-17 03:17:10 +0000
@@ -90,7 +90,7 @@
  * in any case, please use theme('filefield_file') instead.
  */
 function theme_filefield_item($file, $field) {
-  if (filefield_view_access($field['field_name']) && filefield_file_listed($file, $field)) {
+  if (filefield_view_access($field['field_name'], node_load($file['nid'])) && filefield_file_listed($file, $field)) {
     return theme('filefield_file', $file);
   }
   return '';

